libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyParser.java

```diff
@@ -143,6 +150,14 @@ protected Entitlement parseEntitlement(String scopeName, String entitlementType)
         }
     }
 
+    static String getEntitlementClassName(String entitlementType) {
```

Comment:
Huh, this is pretty clever, but something feels odd about allowing the policy files to specify any class they want (though the fact that you put the mandatory `Entitlement` suffix on there helps).

There's also a performance problem here, where we look up the entitlement class by name, scan it for annotations, and reflect on its constructor arguments every time we encounter it in a policy file. Perhaps we could memoize those the first time we find them, so we do the reflection at most once per entitlement class. I suppose this is an optimization we could do later on.

Anyway, no need to change this if you like it the way it is. Just thinking out loud.

Reply:
This also allows multiple spellings of the same entitlement, like `createClassloader` and `create_classloader`.

Reply:
You are probably right -- but we want to discuss it in a follow up I think (definitely with Jack, who decided for this approach). It could be as simple as an allowlist of names though.

> This also allows multiple spellings of the same entitlement, like createClassloader and create_classloader.

Yes, I'm really on the fence for this - if we have an allowlist of names it would limit this case.
For now, I'm thinking of changing it to just look for `_` (if that's what we want to use as a separator). 
Wdyt?